src/service/poc_lora.proto

```diff
@@ -180,8 +186,16 @@ message gateway_reward_share {
   uint64 start_period = 4;
   /// Unix timestamp in seconds of the end of the reward period
   uint64 end_period = 5;
+  /// type of reward
+  reward_type reward_type = 6;
   /// Amount credited to the hotspot for data transfer
-  uint64 dc_transfer_amount = 6;
+  uint64 dc_transfer_amount = 7;
```

Comment:
I'm not sure I like this approach since you now have both a type and a separate amount for each. Why not rename dc_transfer_amount to amount and just have reward_type indicate who that reward goes to and what the amount "means/unit"?

Reply:
dc_transfer amount is not related to these current changes for non gateway rewards.  The proto already defines the following:
beacon_amount
witness_amount
dc_transfer_amount

So we then need a new field or fields to cover the non gateway rewards ( operational fund, stakers and oracles ).  The choice here was to have either a single field to cover all 3 or a distinct field for each.   I started with a singular field to cover the 3 new values,  however as we already have a convention whereby we have specific fields per reward type ie  beacon rewards, witness rewards and dc transfer rewards i felt reusing the one field for a subset of rewards was a poor convention.

Reply:
The enum provides a simple means for the code ( indexer ) to identify which fields to use without relying on checking for non zero values for each `_amount` field

Reply:
bah.. not a great design choice at the time to have separate amounts there in hindsight. 

If we have separate amounts then we wouldn't need the reward_type at all do we?

Reply:
> The enum provides a simple means for the code ( indexer ) to identify which fields to use without relying on checking for non zero values for each `_amount` field

Right but we have to do that for the other amounts already?